src/tools/run-make-support/src/lib.rs

```diff
@@ -169,6 +169,28 @@ pub fn cwd() -> PathBuf {
     env::current_dir().unwrap()
 }
 
+// FIXME(Oneirical): This will no longer be required after compiletest receives the ability
+// to manipulate read-only files. See https://github.com/rust-lang/rust/issues/126334
+/// Ensure that the path P is read-only while the test runs, and restore original permissions
+/// at the end so compiletest can clean up.
+#[track_caller]
+pub fn test_while_readonly<P: AsRef<Path>, F: FnOnce() + std::panic::UnwindSafe>(
+    path: P,
+    closure: F,
+) {
+    let path = path.as_ref();
+    let metadata = fs_wrapper::metadata(&path);
+    let original_perms = metadata.permissions();
+
+    let mut new_perms = original_perms.clone();
+    new_perms.set_readonly(true);
+    fs_wrapper::set_permissions(&path, new_perms);
+
+    assert!(std::panic::catch_unwind(closure).is_ok());
```

Comment:
Here if the closure fails and a panic is caught, then the line below won't be executed, so this doesn't really help :sweat_smile: We should execute the `set_permissions` call always, and then only unwrap the result from the closure after that.

Reply:
Hehe, that was a really silly mistake. Trying this instead:

```rs
    let success = std::panic::catch_unwind(closure).is_ok());

    fs_wrapper::set_permissions(&path, original_perms);
    assert!(success);
```

Reply:
Better, but I would suggest `.unwrap()` instead. Because in that way, we will see the actual error that caused the panic in the console. With this you will see only `assert!(false)`, without knowing what caused the problem :)

Reply:
```rs
    let success = std::panic::catch_unwind(closure);

    fs_wrapper::set_permissions(&path, original_perms);
    success.unwrap();
```

Neat!